docs/templates.md

```diff
@@ -0,0 +1,151 @@
+
+The following keywords can be used in `jj log`/`jj obslog` templates.
+
+* `description: String`
+* `change_id: ChangeId`
+* `commit_id: CommitId`
+* `author: Signature`
+* `committer: Signature`
+* `working_copies: String`: For multi-workspace repository, indicate
+  working-copy commit as `<workspace name>@`.
+* `current_working_copy: Boolean`: True for the working-copy commit of the
+  current workspace.
+* `branches: String`
+* `tags: String`
+* `git_refs: String`
+* `git_head: String`
+* `divergent: Boolean`: True if the change id corresponds to multiple visible
+  commits.
+* `conflict: Boolean`: True if the commit contains merge conflicts.
+* `empty: Boolean`: True if the commit modifies no files.
+
+### Operation keywords
+
+The following keywords can be used in `jj op log` templates.
+
+* `current_operation: Boolean`
+* `description: String`
+* `id: OperationId`
+* `tags: String`
+* `time: TimestampRange`
+* `user: String`
+
+## Operators
+
+The following operators are supported.
+
+* `x.f()`: Method call.
+* `x ++ y`: Concatenate `x` and `y` templates.
+
+## Global functions
+
+The following functions are defined.
+
+* `label(label: Template, content: Template) -> Template`: Apply label to
+  the content. The `label` is evaluated as a space-separated string.
+* `if(condition: Boolean, then: Template[, else: Template]) -> Template`:
+  Conditionally evaluate `then`/`else` template content.
+* `concat(content: Template...) -> Template`:
+  Same as `content_1 ++ ... ++ content_n`.
+* `separate(separator: Template, content: Template...) -> Template`:
+  Insert separator between non-empty contents.
+
+## Types
+
+### Boolean type
+
+No methods are defined.
+
+### CommitId / ChangeId type
+
+The following methods are defined.
+
+* `.short([len: Integer]) -> String`
+* `.shortest([min_len: Integer]) -> ShortestIdPrefix`: Shortest unique prefix.
+
+### Integer type
+
+No methods are defined.
+
+### OperationId type
+
+The following methods are defined.
+
+* `.short([len: Integer]) -> String`
+
+### ShortestIdPrefix type
+
+The following methods are defined.
+
+* `.prefix() -> String`
+* `.rest() -> String`
+* `.upper() -> ShortestIdPrefix`
+* `.lower() -> ShortestIdPrefix`
+
+### Signature type
+
+The following methods are defined.
+
+* `.name() -> String`
+* `.email() -> String`
+* `.username() -> String`
+* `.timestamp() -> Timestamp`
+
+### String type
+
+A string can be implicitly converted to `Boolean`. The following methods are
+defined.
+
+* `.contains(needle: Template) -> Boolean`
+* `.first_line() -> String`
+* `.upper() -> String`
+* `.lower() -> String`
+
+### Template type
+
+Any types can be implicitly converted to `Template`. No methods are defined.
+
+### Timestamp type
+
+The following methods are defined.
+
+* `.ago() -> String`: Format as relative timestamp.
+
+### TimestampRange type
+
+The following methods are defined.
+
+* `.start() -> Timestamp`
+* `.end() -> Timestamp`
+* `.duration() -> String`
+
+## Aliases
```

Comment:
I would:

- Rename this section to `Config`
- Move this entire section to where the sentence about the default templates currently is, in the very beginning of the doc.
- Make that sentence its own paragraph within this section.

Reply:
Done, but moved to the bottom. The examples section will follow (like revsets.md), and I think it's better to place the config section close to it.